### ISSUE TITLE
Create React component for displaying alerts

### ISSUE THREAD

**Issue by oskarhinc** [benetech/MathShare#278]
Create React component for displaying alerts, and attach whole logic related to displaying those messages. (usages of 'DisplayMessage' method - e. g. when trying to calculation-arithmetic expression, while using 'Calc' button)

**Comment by kkrawczyk123** [comment]
@johnhbenetech, I have a question about the testing approach to Screen Reader feature, should I use some browsers plugins? If so what plugins should be supported for each browser?

**Comment by johnhbenetech** [comment]
@sueannma feel free to chime in here as well. Here are some of the most popular combinations:

* Firefox-Chrome / NVDA - https://www.nvaccess.org/
* Safari / VoiceOver - https://help.apple.com/voiceover/info/guide/10.13/
--lower priority--
* Chrome / ChromeVox - http://www.chromevox.com/

We will also need to test JAWS and Dolphin screenreaders, but they require expensive licenses. I will talk with our team to see how we can get you access.

**Comment by sueannma** [comment]
Firefox and Edge with JAWS could also be something we want to check.

As far as I know, Dolphin is more of a reading app, so we probably don't need to worry about that now.

**Comment by kkrawczyk123** [comment]
Currently, I am able to check only 2 of alerts and functionalities causing the rest of the alerts is not ready yet, testing will be continued after functionalities will be added. 
I have also compared in general ChromeVox and VoiceOver readers behavior on qa instance and our branch and it seems to be similar.

**Comment by kkrawczyk123** [comment]
@oskarhinc it would be also good to improve discard dialog message which looks like 
![screenshot from 2018-07-25 14-01-29](https://user-images.githubusercontent.com/33342488/43199782-6ed313ea-9013-11e8-93f9-0548e95f8521.png)
as other dialogs are already improved.

**Comment by oskarhinc** [comment]
@kkrawczyk123 I agree, we could improve the look of the confirmation dialogs. We haven't done it yet because it's a different type of alerts, and we need to find a proper library for this case, so there won't be problems with accessibility. I'll create a separate issue for this improvement tomorrow.

**Comment by kkrawczyk123** [comment]
@oskarhinc, thank you!

**Comment by oskarhinc** [comment]
@kkrawczyk123 I've created the issue: https://github.com/benetech/MathShare/issues/338

**Comment by kkrawczyk123** [comment]
@johnhbenetech I have a question connected to alert display. All alerts appear correctly but they are visible for a very short time and they disappear. I believe that this is a cause of that they are considered in a different way by VoiceOver and ChromeVox screen readers. ChromeVox reads all alerts even if they have already disappeared, good example here is "Please provide a description of your work” that VoiceOver reads the content of the comment field "Use microphone..." and in meantime, alert disappears so it is missed by a screen reader. I would like to suggest leaving alerts until the user clicks on them - it should improve a cooperation with screen readers and also can empower an awareness of alerts existence so users won't miss them. I believe it can be compliant with accessibility which is very important in this project.

**Comment by johnhbenetech** [comment]
@kkrawczyk123 That would be interesting to test, but I do believe there may be a better approach (but it requires some research from the dev team).

I think that ARIA has two options to explore, but these may have compatibility issues with certain reader/browser combinations. There are some example codepens on the below sites that could save testing time.

* ARIA alert role, I believe this could be a voiced-only version of an alert, that essentially interrupts any current voicing, and shouldn't (maybe?) be dependent on the visual alert.
 https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Techniques/Using_the_alert_role

* ARIA Live elements. These seem to be a more flexible way to voice certain alerts. The main difference I see is it has certain 'politeness' settings that will wait for existing voicing to complete before voicing the alert. I think this won't be as relevant as we probably want the user to get the alert immediately.
https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Live_Regions#Simple_Live_Regions

**Comment by sueannma** [comment]
I'm not sure if this went through in the last round of design mocks, but the hope was actually to focus on the annotations field when a user had not provided a description of their work. This might actually get around the issue of the alert being too short or the user having to go through an additional click to discard the message. There would probably need to be some logic built in for the screen reader so that one could escape from reading the full alert (I would expect this to be a standard option for screen readers but am not sure?), but ultimately, this would direct the user to the next mandatory step.
